src/Tribe/Aggregator/Record/Abstract.php

```diff
@@ -301,6 +308,21 @@ public function prep_post_args( $type, $args, $meta = array() ) {
 
 		// prefix all keys
 		foreach ( $meta as $key => $value ) {
+			// skip arrays that are empty
+			if ( is_array( $value ) && empty( $value ) ) {
+				continue;
+			}
+
+			// trim scalars
+			if ( is_scalar( $value ) ) {
+				$value = trim( $value );
+			}
+
+			// if the value is blank or null, let's avoid inserting it
+			if ( null === $value || '' === $value ) {
```

Comment:
Could `empty()` be a better choice here? Might also let us ditch one `if()` block.